fms2_io/netcdf_io.F90

```diff
@@ -1503,86 +1503,80 @@ function get_valid(fileobj, variable_name) &
       add_offset = 0._real64
     endif
 
-	 !Max and and min data values are defined by the valid_range, valid_min, and valid_max attributes if they are present. 
-	 !If the fill_value attribute is present and valid_range is not, then fill_value determines valid_data values. 
-	 !Otherwise, the missing_value attribute determines valid data_values if it is present, and valid_range fill_value attributes are not.
+	!Max and and min data values are defined by the valid_range, valid_min, and valid_max attributes if they are present. 
+    !Assign max/min values using "valid" attributes.
+    if (attribute_exists(fileobj%ncid, varid, "valid_range")) then
+      call get_variable_attribute(fileobj, variable_name, "valid_range", buffer)
+      valid%max_val = buffer(2)*scale_factor + add_offset
+      valid%has_max = .true.
+      valid%min_val = buffer(1)*scale_factor + add_offset
+      valid%has_min = .true.
+    else
+      if (attribute_exists(fileobj%ncid, varid, "valid_max")) then
+        call get_variable_attribute(fileobj, variable_name, "valid_max", buffer(1))
+        valid%max_val = buffer(1)*scale_factor + add_offset
+        valid%has_max = .true.
+      endif
+      if (attribute_exists(fileobj%ncid, varid, "valid_min")) then
+        call get_variable_attribute(fileobj, variable_name, "valid_min", buffer(1))
+        valid%min_val = buffer(1)*scale_factor + add_offset
+        valid%has_min = .true.
+      endif
+    endif
+    valid%has_range = valid%has_min .or. valid%has_max
+
 
-    !Get default max/min from missing_value.  These could be overwritten by
-    !vaild_range/valid_min/valid_max/_Fill_Value.
     if (attribute_exists(fileobj%ncid, varid, "missing_value")) then
       call get_variable_attribute(fileobj, variable_name, "missing_value", buffer(1))
       xtype = get_variable_type(fileobj%ncid, varid)
```

Comment:
@GFDL-Eric I think we can delete this xtype = line now.

Reply:
Resolved with 10096f4